ethstorage/storage_manager.go

```diff
@@ -154,20 +174,22 @@ func (s *StorageManager) CommitBlobs(kvIndices []uint64, blobs [][]byte, commits
 		encoded[i] = true
 	}
 
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	metas, err := s.l1Source.GetKvMetas(kvIndices, s.localL1)
+	metas, err := s.getKvMetas(kvIndices)
 	if err != nil {
 		return nil, err
 	}
+
+	s.mu.Lock()
```

Comment:
The KvMetas can be changed after calling line 177 with lock dropped.  We should stick to the previous code logic from line 157 and acquire the lock from `getKvMetas` and to the rest.